Add note about values.yaml mismatch prevention (#3197)
grafana/mimir: docs/sources/operators-guide/deploy-grafana-mimir/getting-started-helm-charts/_index.md

         # empty, disabled.
   ```

   > **Note:** You can use `helm show values grafana/mimir-distributed` to see all configurable parameters for helm install.

>    > However, refrain from copy and paste the whole thing because all parameters default value are already set and you should only

Reviewer: The fact that there are defaults is not a problem. I would just say that the values file can change in backwards incompatible ways between major versions and you would have to merge every change from the new version to keep it working - not just what you actually customized.

Author: Rephrased the note.

>    > **Note:** You can use `helm show values grafana/mimir-distributed` to see all configurable parameters for helm install.
>    > However, refrain from copy and paste the whole thing because all parameters default value are already set and you should only

Reviewer 2: ```suggestion
   > **Note:** To see all of the configurable parameters for a Helm chart installation, use `helm show values grafana/mimir-distributed`.
   > However, refrain from copying and pasting the whole configuration, because all of the parameters’ default values are already set. Only
```

Author: Applied the first-line suggestion. The second line was rephrased as Krajo's suggestion.

>    > **Note:** You can use `helm show values grafana/mimir-distributed` to see all configurable parameters for helm install.
>    > However, refrain from copy and paste the whole thing because all parameters default value are already set and you should only

Reviewer 2: ```suggestion
   > **Note:** To see all of the configurable parameters for a Helm chart installation, use `helm show values grafana/mimir-distributed`.
   > However, refrain from copying and pasting the whole configuration, because all of the parameters’ default values are already set. Only
```

>    > **Note:** You can use `helm show values grafana/mimir-distributed` to see all configurable parameters for helm install.
>    > However, refrain from copy and paste the whole thing because all parameters default value are already set and you should only

Reviewer 2: ```suggestion
   > **Note:** To see all of the configurable parameters for a Helm chart installation, use `helm show values grafana/mimir-distributed`.
   > However, refrain from copying and pasting the whole configuration, because all of the parameters’ default values are already set. Only
```

>    > override the parameter that you want to change. Likewise, `values.yaml` from `operations/helm/charts/mimir-distributed/values.yaml`

Reviewer: Let's be more explicit: if you want to get the values file from github, you should build the URL like this:
`https://github.com/grafana/mimir/blob/mimir-distributed-<chart-version>/operations/helm/charts/mimir-distributed/values.yaml`
e.g.: https://github.com/grafana/mimir/blob/mimir-distributed-3.1.0/operations/helm/charts/mimir-distributed/values.yaml

Because we git tag every release.

Author: For now I put it the direct link to main branch. Or should I put link to particular version?

>    > override the parameter that you want to change. Likewise, `values.yaml` from `operations/helm/charts/mimir-distributed/values.yaml`
>    > must be used only for matching helm chart version because it is not guaranteed to be backward compatible.

Reviewer 2: ```suggestion
   > override the parameter that you want to change.
   > If you want to get the values file from Github, build the URL as follows because we git tag every release:
   > `https://github.com/grafana/mimir/blob/mimir-distributed-<chart-version>/operations/helm/charts/mimir-distributed/values.yaml`.
   > For example, `https://github.com/grafana/mimir/blob/mimir-distributed-3.1.0/operations/helm/charts/mimir-distributed/values.yaml`. You must use a matching Helm chart version, because charts are not guaranteed to be backwards compatible.
```

         # empty, disabled.
   ```

   > **Note:** To see all of the configurable parameters for a Helm chart installation, use `helm show values grafana/mimir-distributed`.
   > If you want to get the values file from Github, build the URL as follows because we git tag every release:
   > `https://github.com/grafana/mimir/blob/mimir-distributed-<chart-version>/operations/helm/charts/mimir-distributed/values.yaml`.

>    > For example, `https://github.com/grafana/mimir/blob/mimir-distributed-3.1.0/operations/helm/charts/mimir-distributed/values.yaml`.

Reviewer 3: nit: can this be an actual link people can click on?

Author: Nice catch. Not sure why I put backtick in the link. 🤔

>    > You must use a matching Helm chart version, because charts are not guaranteed to be backwards compatible between major version.

Reviewer 3: IIRC we don't really guarantee any backwards compatibility of values files with older versions of the chart. Not only for major versions, but for all. I think the biggest potential breaking change is referencing a new named template in e.g. the nginx config (`nginx.nginxConfig.file`). If you use that new config with any old chart, then it will break. So I think we should change this sentence.

```suggestion
   > You must use a matching Helm chart version, because chart values are not guaranteed to be backwards compatible between versions.
```

Is this your understanding too?

Author: Yes, that's true. The thing that breaks here is when a new named template is used on the old chart.

   > You have to merge every change from the new version to keep it

>    > working and not just what you actually customized.

Reviewer 3: I'm not sure I understand this bit. Are we asking them to pull in all changes between values files versions regardless of whether they customized it or now?

So if I had in my values file

```yaml
global:
  podAnnotations:
    x: y
```
and version 4.7.0 of the chart introduced changes to, e.g.

```yaml
nginx:
  someNewOption: 123
```

then they should add this new nginx field to their values file? This doesn't sound right

Author: Come to this after a discussion with Krajo. However, now I feel it is not right. Definitely, it shouldn't be the scenario above. I will remote this last sentence for now.
